applications/source/rabbit-source/src/test/java/org/springframework/cloud/stream/app/source/rabbit/RabbitSourceListenerTests.java

```diff
@@ -122,6 +123,31 @@ public void testOwnConnection() {
 		}
 	}
 
+	/**
+	 * Sends a message on a separate connection.
+	 *
+	 * @param routingKey message routing key
+	 * @param payload message content
+	 * @param bootFactory the auto-configured connection factory used to get connection coordinates from
+	 */
+	private void sendMessageOnSeparateConnection(String routingKey, Object payload, CachingConnectionFactory bootFactory) {
```

Comment:
Yeh, its gross. We can't however create another RabbitTemplate nor CCF as that will disable SB auto-config.